src/Camelot.ViewModels/Implementations/MainWindow/FilePanels/Nodes/FileViewModel.cs

```diff
@@ -45,5 +59,71 @@ public long Size
     {
         _fileSizeFormatter = fileSizeFormatter;
         _fileTypeMapper = fileTypeMapper;
+        _shellIconsCacheService = shellIconsCacheService;
+        _iconsSettingsService = iconsSettingsService;
+    }
+
+    public Bitmap ShellIcon
+    {
+        get
+        {
+            if (!_loadedShellIcon)
+            {
+                var imageModel = _shellIconsCacheService.GetIcon(FullPath);
+                _shellIcon = FromImageModel(imageModel);
+                _loadedShellIcon = true;
+            }
+            return _shellIcon;
+        }
+    }
+
+    private IconsType GetUserSelectedType()
+    {
+        var model = _iconsSettingsService.GetIconsSettings();
```

Comment:
As a user, very confusing. I applied the theme, and didn't understand why nothing happens.

Reply:
True but it's a tradeoff between code complexity and functionality. In the future I'm planning to add automatic restart to avoid that issue. Loading and storing this value on start saves us a lot of effort of loading it during runtime

Reply:
I think Avalonia v11 has different way of handling theme, that support it (at least for themes)